### DISCUSSION TITLE
List of streams

### DISCUSSION THREAD

**Discussion by berk-efe** [fent/node-ytdl-core#1247] [Q&A]
how can i get the list of the youtube video streams like here

![resim](https://github.com/fent/node-ytdl-core/assets/92454444/535773ca-e519-4451-8bf8-baf7283e6e30)

**Comment by nelliott82** [comment]
If I'm understanding the question right, you want to know how to get the list of video streams for a given YouTube video.

There may be another way, but this method will get you most of the way there.

```
  const info = await ytdl.getInfo(url);
  const videoFormats = ytdl.filterFormats(info.formats, 'videoonly');
  for (const video of videoFormats) {
    console.log(video.itag)
  }
```

You could dig a little deeper to know what other values are available on each `video` and create a more comprehensive compilation of values. But the `itag` will tell you what you need to know. Just compare it to this [comprehensive list](https://gist.github.com/sidneys/7095afe4da4ae58694d128b1034e01e2).

I hope that answers your question!
